input/posts/2020-07-08-grpc-reflection-in-net.md

```diff
@@ -0,0 +1,226 @@
+Title: gRPC Server Reflection in the .NET world
+Published: 2020-07-08
+Tags: 
+- .NET
+- gRPC
+- Protocol Buffers
+---
+> gRPC Server Reflection is an optional extension for servers to assist clients in runtime construction of requests without having stub information precompiled into the client.
+> 
+> -- <cite>[gRPC Server Reflection Protocol](https://github.com/grpc/grpc/blob/master/doc/server-reflection.md)</cite>
+
+The primary use case for gRPC Server Reflection is debugging and testing tools. Think cURL or Postman, but for gRPC. There's however another interesting use case, especially in the .NET world where we have code-first support through [protobuf-net.Grpc](https://github.com/protobuf-net/protobuf-net.Grpc).
+
+If you have been using WCF, you are probably familiar with code-first approach, and exposing metadata about the service using MEX or WSDL endpoints. These metadata endpoints could then be used to generate client stubs. Additionally, if you created a standard SOAP service, to be consumed over HTTP, interoperability with other programming languages was also possible by consuming the WSDL document. This is similar to what many of us do today with ASP.NET Core and OpenAPI. We create the controllers and models in C#, generate OpenAPI definition using [Swashbuckle](https://github.com/domaindrivendev/Swashbuckle.AspNetCore) and then generate clients in TypeScript using [Autorest](https://github.com/Azure/autorest).
```

Comment:
Suggestion: "If you've been using WCF you're probably familiar with a code-first approach that exploses metadata about the service using MEX or WSDL endpoints"

Reply:
think you mean `exposes`, not  `exploses` :)